app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.kt

```diff
@@ -0,0 +1,167 @@
+            activity.getString(R.string.no),
+            onPositiveBtnClick,
+            onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String,
+        negativeButtonText: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?
+    ) {
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            positiveButtonText,
+            negativeButtonText,
+            onPositiveBtnClick,
+            onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?,
+        customView: View,
+        cancelable: Boolean
+    ) {
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            activity.getString(R.string.yes),
+            activity.getString(R.string.no),
+            onPositiveBtnClick,
+            onNegativeBtnClick,
+            customView,
+            cancelable
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String?,
+        onPositiveBtnClick: Runnable?,
+        cancellable: Boolean
+    ) {
+        showAlertDialog(
+            activity,
+            title,
+            message,
+            positiveButtonText,
+            onPositiveBtnClick,
+            cancellable
+        )
+    }
+
+    /**
+     * show a dialog
+     * @param activity
+     * @param title
+     * @param message
+     * @param positiveButtonText
+     * @param negativeButtonText
+     * @param onPositiveBtnClick
+     * @param onNegativeBtnClick
+     * @param customView
+     * @param cancelable
+     */
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String? = null,
+        negativeButtonText: String? = null,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?,
+        customView: View? = null,
+        cancelable: Boolean = false
+    ) {
+
+        /* If the custom view already has a parent, there is already a dialog showing with the view
+         * This happens for on resume - return to avoid creating a second dialog - the first one
+         * will still show
+         */
+        if (customView?.parent != null) {
+            return
+        }
+
+        showSafely(activity, AlertDialog.Builder(activity).apply {
+            setTitle(title)
+            setMessage(message)
+            setView(customView)
+            setCancelable(cancelable)
+            positiveButtonText?.let {
+                setPositiveButton(positiveButtonText) { _: DialogInterface, _: Int ->
```

Comment:
as per the previous explanation you can just say `setPositiveButton(it)` here

Reply:
@macgills Thanks for pointing it out!